### ISSUE TITLE
Investigate automated ArgoCD CLI authentication

### ISSUE THREAD

**Issue by IanKWatts** [BCDevOps/developer-experience#4561]
**Describe the issue**
Users of the shared ArgoCD instance cannot use SSO for CLI access.  Look into the possibility of enabling token-based access to allow large teams to more easily manage their GitOps systems.

**What is the Value/Impact?**
Manual steps create a bottleneck for large teams.  Allowing better CLI access will allow teams of all sizes to increase their GitOps adoption.

**What is the plan? How will this get completed?**
Check the latest ArgoCD docs, etc. and see what we can do.

**Identify any dependencies**
n/a

**Definition of done**
- [ ] A conclusion has been reached on whether to further pursue this goal

**Comment by IanKWatts** [comment]
Users of the shared instance of ArgoCD can use the `argocd` CLI by manually creating a token in the UI and using that with each command invocation, but they cannot use the `argocd login` command to log in with SSO.  As a result, they cannot programmatically create new tokens, because the tokens that they connect with only have permissions for Applications - creating a token requires permission to modify a role.
However, our "gitops alliance" users, which have a grouping of projects, could use the central project to create a token and programmatically create Applications within that project, which has permission to manage resources in any of its namespaces.
Will wait on user feedback before closing this ticket.

**Comment by IanKWatts** [comment]
I have spoken with AG and the argocd CLI will not work for them, because they want to create Applications in each new AppProject, not in a central project.  I'll create a new ticket that reflects the desired outcome.
